model_fitting/module_wraper.py

```diff
@@ -132,15 +267,19 @@ def build_classifier(first_phase_output_path, motif_inference_output_path,
             else:
                 aggregated_values_path = os.path.join(classification_output_path, bc, f'{bc}_values.csv')
                 aggregated_hits_path = os.path.join(classification_output_path, bc, f'{bc}_hits.csv')
-                all_cmds_params.append([meme_path, scanning_dir_path, bc, aggregated_values_path,
-                                        aggregated_hits_path, samplename2biologicalcondition_path, done_path])
+                cmd_merge = [meme_path, scanning_dir_path, bc, aggregated_values_path,
+                    aggregated_hits_path, samplename2biologicalcondition_path, done_path, f'--rank_method {rank_method}']                        
+                if get_sample_for_label:
+                    sample_to_label =','.join(multi_experiments_dict['runs'][exp_name]['biological_motifs_combain'][bc])
+                    cmd_merge += [f'--sample_to_label {sample_to_label}']
+                all_cmds_params.append(cmd_merge)
         else:
             logger.debug(f'skipping score aggregation as {done_path} found')
             num_of_expected_results += 1
 
     if len(all_cmds_params) > 0:
-        executable = 'python' if is_pval else None
-        script_path = f'{src_dir}/model_fitting/{script_name}' if is_pval else f'{src_dir}/tfidf/tfidf'
+        executable = 'python' if use_merge_pvalues else None
+        script_path = f'{src_dir}/model_fitting/{script_name}' if use_merge_pvalues else f'{src_dir}/tfidf/tfidf'

```

Comment:
Are you sure shuffle should be via merge_pvalues and not tfidf?
If there was a change, why not all through merge_pvalue?

Reply:
I made a change in the merge file for support pval and shuffles. 
and because we don't use tfidf I left it like this